recipe-server/client/control_new/components/forms/TrimWhitespaceFormItem.js

```diff
@@ -0,0 +1,34 @@
+import autobind from 'autobind-decorator';
+import { message } from 'antd';
+import React from 'react';
+
+import FormItem from 'control_new/components/forms/FormItem';
+
+@autobind
+export default class TrimWhitespaceFormItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.shouldNotify = true;
+  }
+
+  trimValue(event) {
+    const trimmed = event.target.value.trim();
+    if (trimmed !== event.target.value && this.shouldNotify) {
+      this.shouldNotify = false;
+      message.info(
+        'Whitespace was automatically trimmed from the Preference Name.',
```

Comment:
I almost don't really think any of the `message` stuff is necessary. ♫ It's not unusual ♪ for form fields to auto-trim, especially in this case where we're using it on a field that is whitespace-sensitive.

Regardless, `from the Preference Name` should not be hardcoded here, since this component could be used for other fields.